apps/studio/src/features/editing-experience/components/EditPageDrawer.tsx

```diff
@@ -9,45 +9,48 @@ import {
 import { MdOutlineDragIndicator } from 'react-icons/md'
 import { BsPlus } from 'react-icons/bs'
 import { Button } from '@opengovsg/design-system-react'
-import RootStateDrawer, { Block } from './RootStateDrawer'
-
-export type RootDrawerState = {
-  state: 'root'
-  blocks: Block[]
-}
-
-type AddNewBlockState = {
-  state: 'addBlock'
-}
-
-type NativeEditorState = {
-  state: 'nativeEditor'
-}
-
-type ComplexEditorState = {
-  state: 'complexEditor'
-}
-
-type DrawerState =
-  | RootDrawerState
-  | AddNewBlockState
-  | NativeEditorState
-  | ComplexEditorState
+import RootStateDrawer from './RootStateDrawer'
+import { type DrawerState } from '~/types/editorDrawer'
+import { useEditorDrawerContext } from '~/contexts/EditorDrawerContext'
+import ComponentSelector from '~/components/PageEditor/ComponentSelector'
+import TipTapComponent from './TipTapComponent'
 
 type EditPageDrawerProps = {
   isOpen: boolean
   state: DrawerState
 }
 
-export const EditPageDrawer = ({
-  isOpen: open,
-  state,
-}: EditPageDrawerProps) => {
-  const [currState, setCurrState] = useState<DrawerState>(state)
+export function EditPageDrawer({ isOpen: open, state }: EditPageDrawerProps) {
+  const { drawerState: currState, setDrawerState: setCurrState } =
+    useEditorDrawerContext()
```

Comment:
looks abit off to me, the `state` is at once derived from the parent and the context?

Reply:
yeah whoops this was a hack to just get the state changes working, let me move it out

Reply:
Modified upstream in https://github.com/opengovsg/isomer/pull/219